web/src/main/java/se/inera/intyg/webcert/web/service/unansweredcommunication/UnansweredCommunicationServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+import static se.inera.intyg.webcert.persistence.arende.model.ArendeAmne.KOMPLT;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.clinicalprocess.healthcond.certificate.getcertificateadditions.v1.StatusType;
+import se.inera.intyg.webcert.persistence.arende.model.Arende;
+import se.inera.intyg.webcert.persistence.arende.model.ArendeAmne;
+import se.inera.intyg.webcert.persistence.model.Status;
+import se.inera.intyg.webcert.web.service.arende.ArendeService;
+import se.inera.intyg.webcert.web.web.controller.internalapi.dto.UnansweredCommunicationRequest;
+import se.inera.intyg.webcert.web.web.controller.internalapi.dto.UnansweredCommunicationResponse;
+import se.inera.intyg.webcert.web.web.controller.internalapi.dto.UnansweredQAs;
+
+@Service
+public class UnansweredCommunicationServiceImpl implements UnansweredCommunicationService {
+
+    private final ArendeService arendeService;
+    private static final String FK = "FK";
+
+    public UnansweredCommunicationServiceImpl(ArendeService arendeService) {
+        this.arendeService = arendeService;
+    }
+
+    @Override
+    public UnansweredCommunicationResponse get(UnansweredCommunicationRequest request) {
+        final var maxDaysOfUnansweredCommunication = request.getMaxDaysOfUnansweredCommunication();
+        final var patientIds = request.getPatientIds();
+
+        if (patientIds == null || patientIds.isEmpty() || maxDaysOfUnansweredCommunication == null) {
+            throw new IllegalArgumentException(String.format(
+                "Request is missing parameters required to make request. patientIds '%s' maxDaysOfUnansweredCommunication '%s'",
+                patientIds,
+                maxDaysOfUnansweredCommunication));
+        }
+
+        final var arendenForPatients = arendeService.getArendenForPatientsWithTimestampAfterDate(
+            formatPatientIds(patientIds),
+            LocalDateTime.now().minusDays(maxDaysOfUnansweredCommunication)
+        );
+
+        if (arendenForPatients.isEmpty()) {
+            return new UnansweredCommunicationResponse(Collections.emptyMap());
+        }
+
+        return new UnansweredCommunicationResponse(createUnansweredCommunication(arendenForPatients));
+    }
+
+    private static List<String> formatPatientIds(List<String> patientIds) {
+        return patientIds.stream()
+            .map(patientId -> patientId.replace("-", ""))
+            .collect(Collectors.toList());
+    }
+
+    private Map<String, UnansweredQAs> createUnansweredCommunication(List<Arende> arenden) {
+        final var unansweredCommunication = new HashMap<String, UnansweredQAs>();
+        arenden.forEach(arende -> getUnansweredCommunication(arende, unansweredCommunication));
```

Comment:
You could consider implementing this with streams and separate filter and creating the map. Could make the implementation easier to read.